test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -266,26 +294,182 @@ func responsesConfigMap(name string, namespace *corev1.Namespace) *corev1.Config
+			"AfterControlPlaneInitialized-preloadedResponse": `{"Status": "Success"}`,
+			"AfterClusterUpgrade-preloadedResponse":          `{"Status": "Success"}`,
 		},
 	}
 }
 
-func checkLifecycleHooks(ctx context.Context, c client.Client, namespace string, clusterName string, hooks map[string]string) error {
-	configMap := &corev1.ConfigMap{}
-	configMapName := clusterName + "-hookresponses"
-	err := c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: configMapName}, configMap)
-	Expect(err).ToNot(HaveOccurred(), "Failed to get the hook response configmap")
-	for hook := range hooks {
-		if _, ok := configMap.Data[hook+"-called"]; !ok {
-			return errors.Errorf("hook %s call not recorded in configMap %s/%s", hook, namespace, configMapName)
+// Check that each hook in hooks has been called at least once by checking if its actualResponseStatus is in the hook response configmap.
+// If the provided hooks have both keys and values check that the values match those in the hook response configmap.
+func checkLifecycleHookResponses(ctx context.Context, c client.Client, namespace string, clusterName string, expectedHookResponses map[string]string) error {
+	responseData := getLifecycleHookResponsesFromConfigMap(ctx, c, namespace, clusterName)
+	for hookName, expectedResponse := range expectedHookResponses {
+		actualResponse, ok := responseData[hookName+"-actualResponseStatus"]
+		if !ok {
+			return errors.Errorf("hook %s call not recorded in configMap %s/%s", hookName, namespace, clusterName+"-hookresponses")
+		}
+		if expectedResponse != "" && expectedResponse != actualResponse {
+			return errors.Errorf("hook %s was expected to be %s in configMap got %s", expectedResponse, hookName, actualResponse)
+		}
+	}
+	return nil
+}
+
+// Check that each hook in expectedHooks has been called at least once by checking if its actualResponseStatus is in the hook response configmap.
+func checkLifecycleHooksCalledAtLeastOnce(ctx context.Context, c client.Client, namespace string, clusterName string, expectedHooks []string) error {
+	responseData := getLifecycleHookResponsesFromConfigMap(ctx, c, namespace, clusterName)
+	for _, hookName := range expectedHooks {
+		if _, ok := responseData[hookName+"-actualResponseStatus"]; !ok {
+			return errors.Errorf("hook %s call not recorded in configMap %s/%s", hookName, namespace, clusterName+"-hookresponses")
 		}
 	}
 	return nil
 }
+
+func getLifecycleHookResponsesFromConfigMap(ctx context.Context, c client.Client, namespace string, clusterName string) map[string]string {
+	configMap := &corev1.ConfigMap{}
+	configMapName := clusterName + "-hookresponses"
+	Eventually(func() error {
+		return c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: configMapName}, configMap)
+	}).Should(Succeed(), "Failed to get the hook response configmap")
+	return configMap.Data
+}
+
+// beforeClusterCreateTestHandler provides an blockedCondition function which returns false if the Cluster has entered ClusterPhaseProvisioned.
```

Comment:
I think that's only what `beforeClusterCreateTestHandler` does on top of calling `runtimeHookTestHandler` but it doesn't describe the whole scope of the func right now
(similar below)

But to avoid duplicating the comment from `runtimeHookTestHandler` maybe just change it a bit to say that we are calling `runtimeHookTestHandler` with the blockedCondition func which ..